bin/akula-rpc.rs

```diff
@@ -43,15 +124,572 @@ where
             .unwrap_or(BlockNumber(0)))
     }
 
+    async fn call(&self, call_data: CallData, block_number: BlockNumber) -> RpcResult<Bytes> {
+        let block_hash = canonical_hash::read(&self.db.begin().await?, block_number)
```

Comment:
Here and everywhere: you need to open one and only one read database transaction and use it for the request from start to finish.